R/ensure-format.R

```diff
@@ -141,8 +145,23 @@ get_write_function <- function(format, compression, chunk_size = NULL) {
     return(function(...) fst::write_fst(..., compress = level))
   } else if (format == "csv") {
     return(function(...) readr::write_csv(...))
-    stop("Unsupported format: ", format, call. = FALSE)
+  } else if (format == "json") {
+    fun <- function(x, path, ...) {
+      if (compression == "gzip") {
+        con <- gzfile(path, open = "wb")
+      } else {
+        con <- file(path, open = "w")
+      }
+      on.exit(close(con))
+      x <- as.data.frame(x)
+      # TODO remove after data fixed; this makes stored dims not match
+      # remove null-type columns
+      x <- x[vapply(x, class, character(1L)) != 'vctrs_unspecified']
+      jsonlite::stream_out(as.data.frame(x), con = con, na = "null")
```

Comment:
Probably worth doing here too
```suggestion
      jsonlite::stream_out(as.data.frame(x), con = con, na = "null", verbose = FALSE)
```

Reply:
Unless it speeds it up, I'd rather leave this one. Writing out fanniemae took hours, and this is the only way you can tell if it's hanging. The reading one is also slow, but because that gets captured you can't tell where it is anyway, so there's less point in keeping it. 

Long-term we need to replace jsonlite here with something faster.